### PR TITLE
Documentation: add search caveat when using MariaDB

### PR DESCRIPTION
**Pull request by paviro**
## Proposed change

My shot at trying to making the downsides of using MariaDB clearer.

Closes #(issue or discussion)
[Discussion #8396](https://github.com/paperless-ngx/paperless-ngx/discussions/8396)

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [ ] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [x] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docs/advanced_usage.md [hunk 1]
```diff
@@ -585,10 +585,13 @@ services:
 ### Case Sensitivity
 
 The database interface does not provide a method to configure a MySQL
-database to be case sensitive. This would prevent a user from creating a
+database to be case sensitive. This causes a number of caveats when using MySQL.
+
+#### Tags
+A case **in**sensitive database prevents a user from creating a
```
Comment (shamoon) [inline docs/advanced_usage.md:591]:
```suggestion
database to be case sensitive. A case **in**sensitive database prevents a user from creating a
```

File: docs/advanced_usage.md [hunk 2]
```diff
@@ -585,10 +585,13 @@ services:
...
 tag `Name` and `NAME` as they are considered the same.
 
-Per Django documentation, to enable this requires manual intervention.
+Per Django documentation, mitigating this requires manual intervention.
```
Comment (shamoon) [inline docs/advanced_usage.md:594]:
```suggestion
However, there is a downside to turning on case sensitivity, as it also makes searches case sensitive.   
This means that a document with the title `Invoice` won't be found when searching for `invoice`.

Per Django documentation, making a database case-sensitive requires manual intervention.
```

File: docs/advanced_usage.md [hunk 3]
```diff
@@ -604,6 +607,12 @@ existing tables) with:
     Using mariadb version 10.4+ is recommended. Using the `utf8mb3` character set on
     an older system may fix issues that can arise while setting up Paperless-ngx but
     `utf8mb3` can cause issues with consumption (where `utf8mb4` does not).
+    
+#### Search
+However, there is a downside to turning on case sensitivity, as it also makes searches case sensitive.   
+This means: A document called `Invoice` won't be found when searching for `invoice`.
```
Comment (shamoon) [inline docs/advanced_usage.md:613]:
```suggestion
```

File: docs/configuration.md [hunk 4]
```diff
@@ -60,7 +60,7 @@ database engine. Available options are `postgresql` and
 
     !!! warning
 
-        Using MariaDB comes with some caveats. See [MySQL Caveats](advanced_usage.md#mysql-caveats).
+        Using MariaDB comes with some caveats and is generally not recommended. See [MySQL Caveats](advanced_usage.md#mysql-caveats).
```
Comment (shamoon) [inline docs/configuration.md:63]: Dont add this, its enough to say caveats

**Review by shamoon — CHANGES_REQUESTED** [review]
Thanks, this is generally fine but I think it would be good to just make it more compact. Drop the headings (also the "Tags" issue doesn't apply only to tags anyway) and move the search issue up top below the name thing.

File: docs/advanced_usage.md (hunk 1 above)
Comment (paviro) [reply docs/advanced_usage.md:591]: The tags issue would also affect document types and correspondence  I assume?

Comment (shamoon) [reply docs/advanced_usage.md]: Yes, and other things. Its sort of a general point.

File: docs/advanced_usage.md [hunk 5]
```diff
@@ -585,10 +585,13 @@ services:
 ### Case Sensitivity
 
 The database interface does not provide a method to configure a MySQL
-database to be case sensitive. This would prevent a user from creating a
+database to be case sensitive. A case **in**sensitive database prevents a user from creating a
```
Comment (shamoon) [inline docs/advanced_usage.md:588]:
```suggestion
database to be case-sensitive. A case-**in**sensitive database prevents a user from creating a
```

File: docs/advanced_usage.md [hunk 6]
```diff
@@ -585,10 +585,13 @@ services:
...
 tag `Name` and `NAME` as they are considered the same.
 
-Per Django documentation, to enable this requires manual intervention.
+However, there is a downside to turning on case sensitivity, as it also makes searches case sensitive.
```
Comment (shamoon) [inline docs/advanced_usage.md:591]:
```suggestion
However, there is a downside to turning on case sensitivity, as it also makes searches case-sensitive,
```

File: docs/advanced_usage.md [hunk 7]
```diff
@@ -585,10 +585,13 @@ services:
 ### Case Sensitivity
 
 The database interface does not provide a method to configure a MySQL
-database to be case sensitive. This would prevent a user from creating a
+database to be case sensitive. A case **in**sensitive database prevents a user from creating a
 tag `Name` and `NAME` as they are considered the same.
 
-Per Django documentation, to enable this requires manual intervention.
+However, there is a downside to turning on case sensitivity, as it also makes searches case sensitive.   
+This means that a document with the title `Invoice` won't be found when searching for `invoice`.
```
Comment (shamoon) [inline docs/advanced_usage.md:592]:
```suggestion
so for example a document with the title `Invoice` won't be found when searching for `invoice`.
```

**Review by shamoon — CHANGES_REQUESTED** [review]
Tiny nit and also case-sensitive should be hyphenated.

File: docs/advanced_usage.md (hunk 5 above)
Comment (paviro) [reply docs/advanced_usage.md:588]: I thought so but it wasn't in the original so I didn't! :D

**Review by shamoon — APPROVED** [review]
Cheers thanks

**Pull request merged**